types/src/proof/definition.rs

```diff
@@ -516,7 +516,7 @@ where
 
             // Next we take two children at a time and compute their parents.
             let mut children_iter = children_iter.as_slice().chunks_exact(2);
-            while let Some(chunk) = children_iter.next() {
+            for chunk in children_iter.by_ref() {
```

Comment:
Is this just a minor unrelated refactor?

Reply:
clippy picked up on it and was giving me warnings.